### PR TITLE
Vault Transit Engine guide

### LINKED ISSUE #6636: Document Vault Transit Engine
**Description**
Vault Transit Engine API is now supported by the Vault extension and it would help if it were documented

@vsevel, please take care of it, would be great it were done before 1.2.0 Final

CC @lordofthejars

### PR DESCRIPTION
**Pull request by vsevel**
This adds a guide for "Add support for Vault Transit Secret Engine" https://github.com/quarkusio/quarkus/pull/5147
Fixes https://github.com/quarkusio/quarkus/issues/6636

### CODE REVIEW

**Comment by vsevel** [comment]
@lordofthejars @sberyozkin @gsmet hello all, here is transit secret engine guide. please feel free to review and suggest improvements.

File: docs/src/main/asciidoc/vault-transit.adoc [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Working with HashiCorp Vault's Transit Secret Engine
+
+include::./attributes.adoc[]
+:root-token: s.5VUS8pte13RqekCB2fmMT3u2
+:client-token: s.s93BVzJPzBiIGuYJHBTkG8Uw
+:config-file: application.properties
+:extension-status: preview
+:base-guide: link:vault[Vault guide]
```
Comment (gsmet) [inline docs/src/main/asciidoc/vault-transit.adoc]: I fixed the URL here, otherwise the generated html still points to the .adoc.

**Review by gsmet — APPROVED** [review]
Looks perfect, thanks you!

File: docs/src/main/asciidoc/vault-transit.adoc (hunk 1 above)
Comment (vsevel) [reply docs/src/main/asciidoc/vault-transit.adoc]: then, this is probably wrong: https://github.com/quarkusio/quarkus/blame/master/docs/src/main/asciidoc/spring-security.adoc#L144 (that is where I copied the approach from)
and this: https://github.com/quarkusio/quarkus/blame/master/docs/src/main/asciidoc/index.adoc#L59-L60
correct?

Comment (vsevel) [reply docs/src/main/asciidoc/vault-transit.adoc]: should the vault guides be referenced into the index.adoc?

Comment (gsmet) [reply docs/src/main/asciidoc/vault-transit.adoc]: Yes, it's incorrect, I'll go fix it, thanks for pointing it out.

As for `index.adoc`, we don't really maintain that one anymore.

Comment (gsmet) [reply docs/src/main/asciidoc/vault-transit.adoc]: (and it's not used on the website at all)

**Pull request merged**